xbmc/music/infoscanner/MusicInfoScanner.cpp

```diff
@@ -919,6 +923,58 @@ int CMusicInfoScanner::RetrieveMusicInfo(const std::string& strDirectory, CFileI
   return numAdded;
 }
 
+void CMusicInfoScanner::CheckBoxSets(VECALBUMS &albums)
+{
+  /* Strategy: Having scanned the tags of the albums to add to the library,
+   * check for disc subtitles. If an album only has one subtitled disc, we don't
+   * consider it a boxset as its most likely a 'live'version of the album or a 'bonus disc'.
+   * Single disc subtitles are still added to the database so they can be displayed by
+   * skins at appropriate times,  If 'boxset' was added to the albumreleasetype then add the
+   * album regardless of any other rules. */
+
+  std::string m_oldDiscSubtitle;
+  int count;
+  int discno;
+  int old_discno = 0;
+  for(auto& album : albums)
+  {
+    if (album.bCompilation  && !album.bBoxedSet)
+    {
+      album.bBoxedSet = false;
+      continue; // skip compilations that haven't had the boxset tag added to them
+    }
+    //
+    count = 0;
+    for (auto& song : album.songs)
+    {
+      if (!song.strDiscSubtitle.empty())
+      {
+        if (song.strDiscSubtitle != "dummy")
+        {
+          if (song.strDiscSubtitle != m_oldDiscSubtitle)
+          {
+            m_oldDiscSubtitle = song.strDiscSubtitle;
+            count ++;
+            if (count > 2)
+              album.bBoxedSet = true;
+          }
+        }
+        else
+        {
+          discno = song.iTrack >> 16;
+          song.strDiscSubtitle = StringUtils::Format("Disc %i", discno); // create dummy titles
+          if (discno != old_discno)
+          {
+            old_discno = discno;
+            album.bBoxedSet = true;
+          }
+        }
+      }
+      if (count < 3 && !album.bBoxedSet)  // must have more than 2 titled discs to be considered a boxset unless tag was set
+        album.bBoxedSet = false;    }
```

Comment:
This only reached when !album.bBoxedSet anyway, setting does nothing?

Reply:
Yes it does, because bBoxedSet could have been forced to true by setting the tag, in which case, we don't want to set it to false if there are less than 3 discs.  If you take it out, you cannot make a twin disc set into a boxset.  Not sure why you would want to, but someone might.  It also messes up making boxsets from albums where the discs are in separate directories.